Destination/Backup.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Destination;
+use DateTime;
+
+/**
+ * Class Backup
+ *
+ * Abstraction of one backup file.
+ *
+ * @package Zenstruck\BackupBundle\Destination
+ */
+final class Backup
+{
+    /**
+     * @var string
+     */
+    private $key;
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var integer
+     */
+    private $size;
+
+    /**
+     * @var \DateTimeInterface
+     */
+    private $createdAt;
+
+    public function __construct($key, $filename, $size, $createdAt)
+    {
+        $this->key = $key;
+        $this->filename = $filename;
+        $this->size = $size;
+        $this->createdAt = is_integer($createdAt) ? date_timestamp_set(new DateTime(), $createdAt) : $createdAt;
+    }
+
+    /**
+     * @return string
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename()
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @return int
+     */
+    public function getSize()
```

Comment:
I don’t think that that is good idea, on View layer (wether is that Twig template, or maybe console application) formatting of file size should be done by some sort of template function (Twig filter).

Class Backup belongs to Model layer, and it should not be in charge for data representation, it will violate Single responsibility princip. 


> On Sep 11, 2015, at 3:50 PM, Kevin Bond <notifications@github.com> wrote:
> 
> In Destination/Backup.php <https://github.com/kbond/ZenstruckBackupBundle/pull/3#discussion_r39272560>:
> 
> > +    {
> > +        return $this->key;
> > +    }
> > +
> > +    /**
> > +     * @return string
> > +     */
> > +    public function getFilename()
> > +    {
> > +        return $this->filename;
> > +    }
> > +
> > +    /**
> > +     * @return int
> > +     */
> > +    public function getSize()
> Can we have a getHumanReadable() method?
> 
> Also, I think we should use https://github.com/gabrielelana/byte-units <https://github.com/gabrielelana/byte-units> library for converting.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/kbond/ZenstruckBackupBundle/pull/3/files#r39272560>.
>